datafusion/optimizer/src/analyzer/count_wildcard_rule.rs

```diff
@@ -263,7 +263,9 @@ fn rewrite_schema(schema: &DFSchema) -> DFSchemaRef {
         })
         .collect::<Vec<DFField>>();
     DFSchemaRef::new(
-        DFSchema::new_with_metadata(new_fields, schema.metadata().clone()).unwrap(),
+        DFSchema::new_with_metadata(new_fields, schema.metadata().clone())
```

Comment:
Constructing DFSchema's is unfortunately awkward but I don't think that is the fault of this PR